contrib/gearbox/gearbox.go

```diff
@@ -0,0 +1,50 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package gearbox provides tracing functions for APM
+package gearbox
+
+import (
+	"context"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/gogearbox/gearbox"
+)
+
+const (
+	operationName = "http.request"
+)
+
+//Datadog this method should implement at a middleware layer
+func Middleware(ctx gearbox.Context) {
+	method := string(ctx.Context().Method())
+	opts := []ddtrace.StartSpanOption{
+		tracer.SpanType(ext.SpanTypeWeb),
+		tracer.Tag(ext.HTTPMethod, method),
+		tracer.Tag(ext.HTTPURL, ctx.Context().URI().String()),
+		tracer.Measured(),
+	}
+
+	carrier := gearboxContextCarrier{ctx}
+	if spanctx, err := tracer.Extract(carrier); err == nil {
+		opts = append(opts, tracer.ChildOf(spanctx))
+	}
+
+	span, ctxSpan := tracer.StartSpanFromContext(context.Background(), operationName, opts...)
```

Comment:
Hi. I understand your observation. But I need to create to span to that way because I have to need the context with to Span to pass a **custom Context to the gearbox**

Regards